### PR TITLE
Removed many Python code warnings

### PR DESCRIPTION
**Pull request by sven-oly**
Resolved most of the coding style issues reported by PyCharm on the Python files.
Small update to detailed report HTML.

### CODE REVIEW

File: testdriver/testplan.py [hunk 1]
```diff
@@ -4,519 +4,497 @@
...
+
+            if self.progress_interval and test_num % self.progress_interval == 0:
+                formatted_num = '{:,}'.format(test_num)
+                print('Testing %s / %s. %s of %s' % (
+                    self.exec_list[0], self.testScenario, formatted_num, formatted_count), end='\r')
+
+            # Accumulate tests_per_execution items into a single outline
+            if lines_in_batch < tests_per_execution:
+                test_lines.append(json.dumps(test))
+                lines_in_batch += 1
```
Comment (sffc) [inline testdriver/testplan.py:340]: Observation: I don't immediately see where these lines had been before; they seem to be newly added

**Review by sffc — APPROVED** [review]
Spot-checked it; looks like a really great change! Really happy to see a consistent style applied to the .py files.

**Review by echeran — APPROVED** [review]
LGTM. This is a huge improvement in readability, thanks!

File: testdriver/testplan.py (hunk 1 above)
Comment (sven-oly) [reply testdriver/testplan.py:340]: These were changed from the previous version, removing an intermediate variable.

**Pull request merged**